packages/main/src/SegmentedButton.ts

```diff
@@ -260,17 +263,71 @@ class SegmentedButton extends UI5Element {
 		}
 	}
 
+	/**
+	 * Performs the layout for the SegmentedButton component.
+	 *
+	 * Calculates and sets the appropriate width for the component based on the inline style and/or CSS class.
+	 * Adjusts the responsiveness of the component based on the parent width.
+	 *
+	 * @returns {Promise<void>} A Promise that resolves once the layout is completed.
+	 * @private
+	 */
 	async _doLayout(): Promise<void> {
-		const itemsHaveWidth = this.widths && this.widths.some(itemWidth => itemWidth > 2); // 2 pixels added for rounding
+		const rootFontSize = parseFloat(getComputedStyle(document.documentElement).fontSize); // Gets the root font size in pixels in order to convert rem to pixels correctly
+		const itemMinWidth = 2.5; // Item minimum width in rem
+		const itemMinWidthPx = itemMinWidth * rootFontSize; // Converts rem to pixels
+
+		const itemsHaveWidth = this.widths && this.widths.some(itemWidth => itemWidth > 2);
 		if (!itemsHaveWidth) {
 			await this.measureItemsWidth();
 		}
 
 		const parentWidth = this.parentNode ? (this.parentNode as HTMLElement).offsetWidth : 0;
 
-		if (!this.style.width || this.percentageWidthSet) {
-			this.style.width = `${Math.max(...this.widths!) * this.items.length}px`;
-			this.absoluteWidthSet = true;
+		/**
+		 * Calculates the default width for the SegmentedButton component based on the items' widths, if there is no custom width set.
+		 */
+		const defaultComponentWidth = `${Math.max(...this.widths!) * this.items.length}px`;
+		const inlineWidth = this.style.width; // Inline style width
+		const classWidth = this.getAttribute("class") ? getComputedStyle(this).width : ""; // Width defined by CSS class
+
+		/**
+		 * Calculates the width to be set for the SegmentedButton component based on the inline style and/or CSS class.
+		 *
+		 * @returns {string} The calculated width for the SegmentedButton component.
+		 * @private
+		 */
+		const calculateWidth = (): string => {
+			const numItems = this.items.length;
+			let resultWidth = "";
+
+			if (inlineWidth && classWidth) {
+				resultWidth = inlineWidth;
+			} else if (inlineWidth) {
+				resultWidth = inlineWidth;
+			} else if (classWidth) {
+				resultWidth = classWidth;
+			} else {
+				resultWidth = defaultComponentWidth;
+			}
+
+			const widthValue = parseInt(resultWidth.replace("px", ""));
```

Comment:
parseInt(resultWidth) also works. Please change it on all places you have used it.